engine/source/public/core/engine.hpp

```diff
@@ -1,8 +1,14 @@
 ﻿#pragma once
 
+#include "types.hpp"
+
 class Engine
 {
 public:
+    int2 windowExtent {1024, 576};
```

Comment:
Because that is another library, and GLM is a bit out of scope for this PR. I needed a simple way to get a vec2, and this way we could eventually create an internal format that we can expose to the game project without exposing the glm library.

Reply:
Fair, but we should not create our own implementations for everything. Or we could wrap GLM.

Reply:
I agree with wrapping GLM, we'll (re)move this once GLM has been added